src/gtclang/Driver/Driver.cpp

```diff
@@ -30,7 +30,8 @@
 
 namespace gtclang {
 
-int Driver::run(const llvm::SmallVectorImpl<const char*>& args) {
+std::pair<int, std::shared_ptr<dawn::SIR>>
```

Comment:
In general, I'm not a big fan of returning a pair. I think it is much more expressive to return a custom pair which allows to access the members explicitly. For example, I would use something like:
```
struct ReturnValue {
  int ExitCode;
  std::shared_ptr<dawn::SIR> SIR;
};
```
which makes it immediately obvious which member you need to query to get the exit status (instead of figuring out if it is `first` or `second`). Creation of this object is usually even more trivial as you can simply write `return ReturnValue{1, sir_};`.
As an added bonus, you could add an implicit conversion to int which just returns the error code.
```
struct ReturnValue {
  int ExitCode;
  std::shared_ptr<dawn::SIR> SIR;

  operator int() { return ExitCode; }
};
```